pkg/config/draftconfig.go

```diff
@@ -211,6 +225,46 @@ func (d *DraftConfig) VariableMapToDraftConfig(flagVariablesMap map[string]strin
 	}
 }
 
+// SetFileNameOverride sets the filename override for a specific file
+func (d *DraftConfig) SetFileNameOverride(input, override string) {
+	if d.FileNameOverrideMap == nil {
+		d.FileNameOverrideMap = make(map[string]string)
+	}
+	d.FileNameOverrideMap[input] = override
+}
+
+func (d *DraftConfig) DeepCopy() *DraftConfig {
+	newConfig := &DraftConfig{}
+	newConfig.TemplateName = d.TemplateName
+	newConfig.DisplayName = d.DisplayName
+	newConfig.Description = d.Description
+	newConfig.Type = d.Type
+	newConfig.Versions = d.Versions
+	newConfig.DefaultVersion = d.DefaultVersion
+
+	newConfig.Variables = make([]*BuilderVar, len(d.Variables))
+	for i, variable := range d.Variables {
+		newConfig.Variables[i] = variable.DeepCopy()
+	}
+
+	return newConfig
+}
```

Comment:
💡 Observation/Suggestion only please: struct literal to initialize newConfig fields, making the code more compact and readable. Which means we can jus use the d.variable as is, loop remains same.

```*.go
func (d *DraftConfig) DeepCopy() *DraftConfig {
	newConfig := &DraftConfig{
		TemplateName:   d.TemplateName,
		DisplayName:    d.DisplayName,
		Description:    d.Description,
		Type:           d.Type,
		Versions:       d.Versions,
		DefaultVersion: d.DefaultVersion,
		Variables:      make([]*BuilderVar, len(d.Variables)),
	}

	for i, variable := range d.Variables {
		newConfig.Variables[i] = variable.DeepCopy()
	}

	return newConfig
}

```

Reply:
cleaned up all the `DeepCopy` definitions